### PR TITLE
Issue 14878 language web api

### PR DESCRIPTION
**Pull request by wezell**
This fixes issue #14867

### CODE REVIEW

**Review by jgambarios — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java [hunk 1]
```diff
@@ -589,7 +589,7 @@ public InputStream writeObject(final VelocityResourceKey key)
 
         long language = new Long(key.language);
         ContentletVersionInfo info = APILocator.getVersionableAPI().getContentletVersionInfo(key.id1, language);
-        if (info == null && language != defualtLang && LanguageWebAPI.canDefaultContentToDefaultLanguage()) {
+        if (info == null && language != defualtLang && APILocator.getLanguageAPI().canDefaultContentToDefaultLanguage()) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java]: there is a typo here defualtLang

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java [hunk 2]
```diff
@@ -815,4 +816,53 @@ public String getHTML(String uri, Host host, boolean liveMode, String contentId,
...
+	@Override
+    public IHTMLPage findByIdLanguageFallback(final Identifier identifier, final long tryLang, final boolean live, final User user, final boolean respectFrontEndPermissions)
+            throws DotDataException, DotSecurityException {
+
+
+	    return findByIdLanguageFallback(identifier.getId(), tryLang, live, user, respectFrontEndPermissions);
+
+    }
+	
+	@Override
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java]: shouldn't be annotated by CloseDbIfOpened

File: dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPI.java [hunk 3]
```diff
@@ -253,4 +253,12 @@
 	public List<Language> getAvailableContentLanguages(final String contentletInode, final User user)
 			throws DotSecurityException, DotDataException;
 
+    boolean canDefaultContentToDefaultLanguage();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPI.java]: copy the implementation doc to here

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java]: I think so yes

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAPITest.java [hunk 4]
```diff
@@ -309,4 +319,74 @@ public void removeOneLanguageOfHtmlAsset() throws Exception  {
 
 	}
 
+	@DataProvider
```
[the remark this answers is not in the document]
Comment (dsilvam) [reply dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAPITest.java]: codacy does not understand data providers :D

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java (hunk 2 above)
Comment (wezell) [reply dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java]: agreed

**Pull request merged**